code/model/DMSDocument.php

```diff
@@ -1003,6 +1003,19 @@ public function onBeforeWrite()
             }
             $this->LastEditedByID = $currentUserID;
         }
+
+        // make sure default DownloadBehavior is respected when initially writing document
+        // in case the default in the enum is different than what's set in an outside config
+        $defaultDownloadBehaviour = Config::inst()->get('DMSDocument', 'default_download_behaviour');
+        if ($this->DownloadBehavior == null && !empty($defaultDownloadBehaviour)) {
+            $possibleBehaviors = $this->dbObject('DownloadBehavior')
+                ->enumValues();
+
+            $behavior = $possibleBehaviors[$defaultDownloadBehaviour];
```

Comment:
you need an `isset` or `!empty` or `array_key_exists` check here

Reply:
Done :)